ml-algorithms/src/main/java/org/opensearch/ml/engine/tools/SearchIndexTool.java

```diff
@@ -211,5 +213,10 @@ public String getDefaultType() {
         public String getDefaultVersion() {
             return null;
         }
+
+        @Override
+        public Map<String, Object> getDefaultAttributes() {
```

Comment:
To avoid adding this to every tool, how about moving the default implementation to interface class 
```
    default Map<String, Object> getDefaultAttributes() {
        return DEFAULT_ATTRIBUTES;
    }
```